File: tests/audio/soundstream.cpp
#include <iostream>
#include <SFML/Audio.hpp>

class DerivableSoundStream : public sf::SoundStream
{
public :
	DerivableSoundStream();

	//void initialize(unsigned int channelCount, unsigned int sampleRate);
	
protected:
	virtual bool onGetData(sf::SoundStream::Chunk &data);
	virtual void onSeek(sf::Time timeOffset);
};

DerivableSoundStream::DerivableSoundStream():
sf::SoundStream  ()
{
	sf::SoundStream::initialize(1, 44100);
};

bool DerivableSoundStream::onGetData(sf::SoundStream::Chunk &data)
{
	std::cout << "onGetData()" << std::endl;
	return true;
}

void DerivableSoundStream::onSeek(sf::Time timeOffset)
{
	std::cout << "onSeek()" << std::endl;
}

int main()
{
	DerivableSoundStream stream;
	stream.play();
	
	return EXIT_SUCCESS;
}

File: tests/audio/soundstream.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
#
# pySFML2 - Cython SFML Wrapper for Python
# Copyright 2012, Jonathan De Wachter <dewachter.jonathan@gmail.com>
#
# This software is released under the GPLv3 license.
# You should have received a copy of the GNU General Public License
# along with this program.  If not, see <http://www.gnu.org/licenses/>.

import sfml as sf

class CustomStream(sf.SoundStream):
	def __init__(self):
		sf.SoundStream.__init__(self)
		
		self.initialize(1, 44100)
		
	def open(self, location): pass
		
	def on_get_data(self, data):
		return True
		
	def on_seek(self, time_offset):
		pass


mystream = CustomStream()
mystream.play()
input()
